### DISCUSSION TITLE
dnd.startDrag的回调

### DISCUSSION THREAD

**Discussion by io-o** [didi/LogicFlow#1421] [Q&A]
## 问题背景

拖拽节点后，增加回调函数，获取节点dom，增加动画处理，有没有其他方法解决？

## 提案

解决方案，最好可以提供有一些伪代码

## 相关信息

一些关于需求的额外信息和截图

**Comment by wumail** [comment]
没太看懂，要做什么？怎么做的？

**Comment by io-o** [comment]
> 没太看懂，要做什么？怎么做的？

使用 lf.dnd.startDrag生成节点，想给生成的自定义html节点，增加动画效果，在哪里做比较合适？ 在自定义节点上直接加，初始化的时候，也会有动画效果，目前只想在拖拽节点的时候去增加。

**Comment by wumail** [comment]
> > 没太看懂，要做什么？怎么做的？
> 
> 使用 lf.dnd.startDrag生成节点，想给生成的自定义html节点，增加动画效果，在哪里做比较合适？ 在自定义节点上直接加，初始化的时候，也会有动画效果，目前只想在拖拽节点的时候去增加。

具体是什么样的动画呀，你的自定义节点是通过什么方式实现的，inner html? react? vue? 我建议你加交流群在群里交流，这信息反馈有点慢了

**Comment by io-o** [comment]
> > > 没太看懂，要做什么？怎么做的？
> > 
> > 
> > 使用 lf.dnd.startDrag生成节点，想给生成的自定义html节点，增加动画效果，在哪里做比较合适？ 在自定义节点上直接加，初始化的时候，也会有动画效果，目前只想在拖拽节点的时候去增加。
> 
> 具体是什么样的动画呀，你的自定义节点是通过什么方式实现的，inner html? react? vue? 我建议你加交流群在群里交流，这信息反馈有点慢了

HtmlNode继承，原生html实现，动画就是animation实现, 在拖拽节点到画布上生成后，触发动画。 交流群就是qq群么？

**Comment by wumail** [comment]
> > > > 没太看懂，要做什么？怎么做的？
> > > 
> > > 
> > > 使用 lf.dnd.startDrag生成节点，想给生成的自定义html节点，增加动画效果，在哪里做比较合适？ 在自定义节点上直接加，初始化的时候，也会有动画效果，目前只想在拖拽节点的时候去增加。
> > 
> > 
> > 具体是什么样的动画呀，你的自定义节点是通过什么方式实现的，inner html? react? vue? 我建议你加交流群在群里交流，这信息反馈有点慢了
> 
> HtmlNode继承，原生html实现，动画就是animation实现, 在拖拽节点到画布上生成后，触发动画。 交流群就是qq群么？

微信群相对活跃一些，加官方微信：logic-flow，他会拉你进去

**Comment by wumail** [comment]
试试在node:dnd-drag回调里做
![image](https://github.com/didi/LogicFlow/assets/56008486/c1771c1f-1a1a-402c-90f2-5d0dfdd24fa7)
